vavr/src/main/java/io/vavr/collection/LinkedHashMap.java

```diff
@@ -901,7 +901,11 @@ public int size() {
 
     @Override
     public boolean equals(Object o) {
-        return Collections.equals(this, o);
+        if (o instanceof LinkedHashMap) {
+            return Collections.equals(this.list, ((LinkedHashMap<?, ?>) o).list);
+        } else {
+            return Collections.equals(this, o);
+        }
```

Comment:
We changed the equals behavior.

This is not correct according to the `Traversable.equals()` javadoc. We act like Scala, which is:

```java
scala> import scala.collection.immutable._
import scala.collection.immutable._

scala> val map1 = ListMap((1, 'a'), (2, 'b'))
map1: scala.collection.immutable.ListMap[Int,Char] = ListMap(1 -> a, 2 -> b)

scala> val map2 = ListMap((2, 'b'), (1, 'a'))
map2: scala.collection.immutable.ListMap[Int,Char] = ListMap(2 -> b, 1 -> a)

scala> map1 == map2
res0: Boolean = true
```

_Note: The immutable LinkedHashMap is called ListMap in Scala_